docs/docs/propeller/index.md

```diff
@@ -0,0 +1,40 @@
+---
+sidebar_position: 4
+---
+
+# Propeller
+
+Propeller is EquinoxJS's "[propulsion](https://github.com/jet/propulsion)
+lite". It's a set of abstractions for building reactive event processing
+pipelines catering specifically for event sourced reactions from Equinox's
+stores.
+
+## StreamsSink
+
+The `StreamsSink` receives batches of events from a source and intelligently
+manages handling them.
+
+- **Stream management**: maintains an ordered queue of incoming streams,
+  ensuring that reactions processing aligns with the ordering of the triggering
+  events; the aim being to be able to checkpoint attained progress ('tick off
+  batches') as soon as possible.
+- **Concurrent stream handling**: To ensure optimal performance, `StreamsSink`
+  restricts the number of streams that are processing at any given moment to a
+  configurable limit. Additionally guarantees that each stream can have at most
+  one handler actively processing.
+- **Batch merging**: intelligently merges new events with existing streams.
```

Comment:
maybe this goes before the preceding point, leading into the fact that the handler gets the latest coalesced stream events from across batches